Src/HALALMock/Services/Communication/FDCAN/FDCAN.cpp

```diff
@@ -89,43 +92,52 @@ bool FDCAN::transmit(uint8_t id, uint32_t message_id, const char* data, FDCAN::D
 		return false;
 	}
 
-	instance->tx_header.Identifier = message_id;
+    size_t buffer_len = sizeof(message_id)+ sizeof(dlc) + dlc;
+    char* temp_data = new char[buffer_len];
 
-	if (dlc != FDCAN::DLC::DEFAULT) {
-		instance->tx_header.DataLength = dlc;
-	}
+    temp_data[0] = (message_id >> 24);
+    temp_data[1] = (message_id >> 16);
+    temp_data[2] = (message_id >> 8);
+    temp_data[3] = message_id;
 
-	HAL_StatusTypeDef error = HAL_FDCAN_AddMessageToTxFifoQ(instance->hfdcan, &instance->tx_header, (uint8_t*)data);
+	temp_data[4] = ((uint32_t)dlc >> 24);
+	temp_data[5] = ((uint32_t)dlc >> 16);
+	temp_data[6] = ((uint32_t)dlc >> 8);
+	temp_data[7] = (uint32_t)dlc;
 
-	if (error != HAL_OK) {
+	memcpy((temp_data + sizeof(message_id)+ sizeof(dlc)), data, dlc);
```

Comment:
again, dlc is an enum, not a value you can use as the length here, this won't work as you expect because the variant values won't match